### DISCUSSION TITLE
How do attributes propagate through structs?

### DISCUSSION THREAD

**Discussion by verdverm** [cuelang/cue#888] [Q&A] [answered]
In the following example, I am trying to understand how attributes propagate with definitions (though it looks like regular structs have the same results). I've tried several variations across conjunction and embedding.

I expected the attributes to propagate more than they do (for the Bs and Cs). Is this a bug or intended behavior?

### main.cue

```
// FieldAttr on field in def
#A: {
        s: string @input()
}
// DeclAttribute on def
#B: {
        @input()
        s: string
}
// FieldAttr on def
#C: {
        s: string
} @input()

a1: #A
a2: { #A }
b1: #B
b2: { #B }
c1: #C
c2: { #C }
```

### cue eval -A main.cue

```
#A: {
    s: string @input()
}
#B: {
    @input()
    s: string
}
#C: {
    s: string
} @input()
a1: {
    s: string @input()
}
a2: {
    s: string @input()
}
b1: {
    s: string
}
b2: {
    s: string
}
c1: {
    s: string
}
c2: {
    s: string
}
```

**Comment by mpvl** [comment, marked as answer]
It seems that field attributes get propagate and declaration attributes not. If so, that is a bug. All attributes should be propagated.

**Reply by verdverm** [reply]
Filed https://github.com/cuelang/cue/issues/894

**Comment by cueckoo** [comment]
This discussion has been migrated to https://github.com/cue-lang/cue/issues/888.

For more details about CUE's migration to a new home, please see https://github.com/cue-lang/cue/issues/1078.
